cachet_url_monitor/configuration.py

```diff
@@ -41,28 +43,32 @@ class Configuration(object):
     """
 
     def __init__(self, config_file):
-        # TODO(mtakaki#1|2016-04-28): Accept overriding settings using environment
-        # variables so we have a more docker-friendly approach.
         self.logger = logging.getLogger('cachet_url_monitor.configuration.Configuration')
         self.config_file = config_file
         self.data = load(file(self.config_file, 'r'))
 
         # We need to validate the configuration is correct and then validate the component actually exists.
         self.validate()
-        self.headers = {'X-Cachet-Token': self.data['cachet']['token']}
-        self.status = self.get_current_status(self.data['cachet']['component_id'])
 
-        self.logger.info('Monitoring URL: %s %s' %
-                         (self.data['endpoint']['method'], self.data['endpoint']['url']))
-        self.expectations = [Expectaction.create(expectation) for expectation
-                             in self.data['endpoint']['expectation']]
+        self.headers = {'X-Cachet-Token': os.environ.get('CACHET_TOKEN') or self.data['cachet']['token']}
+
+        self.endpoint_method = os.environ.get('ENDPOINT_METHOD') or self.data['endpoint']['method']
+        self.endpoint_url = os.environ.get('ENDPOINT_URL') or self.data['endpoint']['url']
+        self.endpoint_timeout = os.environ.get('ENDPOINT_TIMEOUT') or self.data['endpoint'].get('timeout')
+
+        self.api_url = os.environ.get('CACHET_API_URL') or self.data['cachet']['api_url']
+        self.component_id = os.environ.get('CACHET_COMPONENT_ID') or self.data['cachet']['component_id']
+        self.metric_id = os.environ.get('CACHET_METRIC_ID') or self.data['cachet'].get('metric_id')
+
+        self.status = self.get_current_status(self.api_url, self.component_id, self.headers)
+
+        self.logger.info('Monitoring URL: %s %s' % (self.endpoint_method, self.endpoint_url))
+        self.expectations = [Expectaction.create(expectation) for expectation in self.data['endpoint']['expectation']]
         for expectation in self.expectations:
             self.logger.info('Registered expectation: %s' % (expectation,))
 
-    def get_current_status(self, component_id):
-        get_status_request = requests.get(
-            '%s/components/%d' % (self.data['cachet']['api_url'], self.data['cachet']['component_id']),
-            headers=self.headers)
+    def get_current_status(self, endpoint_url, component_id, headers):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://www.codacy.com/app/mitsuotakaki/cachet-url-monitor/file/2605355958/issues/source?bid=3273968&fileBranchId=3309351#l70)